src/parser/wat-parser.cpp

```diff
@@ -138,7 +138,11 @@ void propagateDebugLocations(Module& wasm) {
       return std::make_unique<Propagator>();
     }
   };
-  PassRunner runner(&wasm);
+  // The parser should not be responsible for validation.
+  PassOptions opts;
+  opts.validate = false;
+  opts.validateGlobally = false;
+  PassRunner runner(&wasm, opts);
```

Comment:
Maybe a better way to get this outcome is to do `setIsNested(true)`. Nested pass runners do not run validation and do not show up on various logging, as they are internal to a pass. (This isn't a pass, but the idea would be the same - it's an internal operation we don't consider as "atomic" in the sense of having valid input and output, and there is no point to log it.)

Reply:
I see a thumbs up here but the code looks unchanged - is this in progress or did you prefer the current form for some reason?

Reply:
Sorry, graphite keeps marking this ready for review, but it's definitely still a draft. I'll make this change before this is ready for real review, but I've been focusing on fixing fuzz bugs before coming back to this.

Reply:
I still think this is worth doing.